python/src/nanoarrow/iterator.py

```diff
@@ -263,6 +263,25 @@ def _binary_iter(self, offset, length):
             for start, end in zip(starts, ends):
                 yield bytes(data[start:end])
 
+    def _decimal_iter(self, offset, length):
+        from decimal import Context, Decimal
+        from sys import byteorder
+
+        storage = self._primitive_iter(offset, length)
+        precision = self._schema_view.decimal_precision
+        scaleb = Decimal(-self._schema_view.decimal_scale)
```

Comment:
Does this need to be converted to a Decimal? 
The docs seem to suggest this can be just an integer (https://docs.python.org/3/library/decimal.html#decimal.Decimal.scaleb)

Also, I might add some comments about what is going on for people not familiar with this method (I had to look it up to understand what you wrote, but certainly makes sense now)